### ISSUE TITLE
`IllegalArgumentException` with `:shrunk` field in custom `IPersistentMap` implementation

### ISSUE THREAD

**Issue by evg-tso** [cursive-ide/cursive#2966]
After a recent upgrade of Cursive (version `1.13.5` and later), we encountered an issue where tests start to fail with the following error:
```
Error: java. lang. IllegalArgumentException: No such field :shrunk - that an item isn't found when it doesn't exist
```

This error occurs when Cursive tries to access the `:shrunk` field via `clojure.lang.RT.get()`. The key issue is that we are using a custom implementation of `IPersistentMap`, which only allows access to a predefined set of keys. The `:shrunk` field is not part of this set, which causes the error.  

This is a Cursive only error, as running `lein test` doesn't fail.  

**Steps to Reproduce**

1. Create a custom implementation of `IPersistentMap` that restricts access to a predefined set of keys.
2. Run tests in Cursive that involve assertions or other actions that might trigger access to the `:shrunk` key.
3. See the `IllegalArgumentException` when Cursive tries to access a non-existent `:shrunk` field.

For a minimal reproduction, please refer to the example repository here: [Repro Example](https://github.com/evg-tso/cursive-shrink-issue).

**Additional Context**

The root cause is detailed in the README within the provided repo. To summarize, Cursive tries to access the `:shrunk` field, but our custom `IPersistentMap` does not support this field, leading to an exception. The issue occurs even though the tests pass successfully when run via `lein test`.

![image](https://github.com/user-attachments/assets/ff229a59-79ac-4a3e-a07a-24ea46359441)

**Comment by cursive-ide** [comment]
There's another similar report in Slack from @simongray here: https://clojurians.slack.com/archives/C0744GXCJ/p1727946413145859

> I’m trying out Datalevin. Unfortunately, Cursive won’t let me view the exception I’m getting. Cursive seems to crawl the exception before outputting it, so I get a runtime exception caused by Cursive calling an unsupported method instead of the ex-info I would like, e.g.

![Skærmbillede 2024-10-03 kl  11 06 10](https://github.com/user-attachments/assets/5e45dec1-bbc3-4118-9624-ea78a6c4c4f4)

**Comment by cursive-ide** [comment]
Fixed, thanks for the great repro case.

**Comment by cursive-ide** [comment]
Fix released in 1.14.1-eap1 on 2024-12-11.

**Comment by cursive-ide** [comment]
Fix released in 2025.1 on 2025-04-16.
